pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
```

Comment:
@kargakis and I changed it back because then the events will show for DC which is wrong... they really are related to the deployment(rc) they are run for.